centreon-certified/prometheus/prometheus-pushgateway-metrics-apiv2.lua

```diff
@@ -0,0 +1,613 @@
+-- create_metric_name: concatenates data to create the metric name
+-- @param {string} label, the name of the perfdata
+-- @param {string} unit, the unit name
+-- @return {string} name, the prometheus metric name (open metric format)
+--------------------------------------------------------------------------------
+function EventQueue:create_metric_name (label, unit)
+  local name = ''
+  local sdesc = 'host'
+  if (self.sc_event.event.service_description) then
+    sdesc = self.sc_event.event.service_description
+  end
+  local hname = self.sc_event.event.cache.host.name
+
+  if (unit ~= '') then
+    if (self.enable_extended_metric_name == 0) then
+      name = label .. '_' .. unit
+    else
+      name = hname .. '_' .. sdesc .. ':' .. label .. '_' .. unit
+    end
+  else
+    if (self.enable_extended_metric_name == 0) then
+      name = label
+    else
+      name = hname .. '_' .. sdesc .. ':' .. label
+    end
+  end
+
+  return convert_to_openmetric(name)
+end
+
+--------------------------------------------------------------------------------
+---- EventQueue:format_metric_service method
+-- @param metric {table} a single metric data
+-------------------------------------------------------------------------------
+function EventQueue:format_metric_event(metric)
+  self.sc_logger:debug("[EventQueue:format_metric]: start real format metric ")
+  local event = self.sc_event.event
+  local type  = self:get_metric_type(metric)
+  local unit  = unit_mapping(metric.uom)
+  local label = metric.metric_name
+  local name  = self:create_metric_name(label, unit)
+  local data = ''
+  local sdesc = event.formated_event.prom_sdesc
+
+  data = '# TYPE ' .. name .. ' ' .. type .. '\n'
+  data = data .. self:add_unit_info(label, unit, name)
+  
+  if not event.hostgroupsLabel then
+    data = data .. name .. '{label="' .. label .. '", host="' .. event.cache.host.name .. '", service="' .. sdesc .. '"} ' .. metric.value .. '\n'
+  else
+    data = data .. name .. '{label="' .. label .. '", host="' .. event.cache.host.name .. '", service="' .. sdesc .. '", ' ..  event.hostgroupsLabel .. '} ' .. metric.value .. '\n'
+  end
+
+  if (self.enable_threshold_metrics == 1) then 
+    data = data .. self:threshold_metrics(metric, label, unit, type)
+  end
+
+  event.formated_event.payload = data
+
+  self:add()
+  self.sc_logger:debug("[EventQueue:format_metric]: end real format metric ")
+end
+
+--------------------------------------------------------------------------------
+-- ifnumber_not_nan: [for Prometheus] check if a number is a number (and not a NaN)
+-- @param {number} number, the number to check
+-- @return {boolean}
+--------------------------------------------------------------------------------
+local function ifnumber_not_nan (number)
+  if (number ~= number) then
+    return false
+  elseif (type(number) ~= 'number') then
+    return false
+  else
+    return true
+  end
+end
```

Comment:
```suggestion
--------------------------------------------------------------------------------
-- ifnumber_not_nan:  check if a number is a number (and not a NaN)
-- @param {number} number, the number to check
-- @return {boolean}
--------------------------------------------------------------------------------
local function is_number_and_not_a_NaN (number)
  if (number ~= number) then
    return false
  end
  
  if (type(number) ~= "number") then
    return false
  end

  return true
end
```

- This function can be useful outside of this stream connector so it must be put in the sc_common lib instead (and documented 😈 
- the suggestion is how I would write it nowadays (i've not updated the comment but i've changed the name of the function because the best practice when creating a function that returns a boolean is to start it with "is_" something...) 
-